src/libbson/src/bson/bson-decimal128.c

```diff
@@ -171,6 +166,7 @@ bson_decimal128_to_string (const bson_decimal128_t *dec, /* IN  */
    /* Decode combination field and exponent */
    combination = (high >> 26) & COMBINATION_MASK;
 
+   int biased_exponent;
    if (BSON_UNLIKELY ((combination >> 3) == 3)) {
       /* Check for 'special' values */
       if (combination == COMBINATION_INFINITY) { /* Infinity */
```

Comment:
```suggestion
      if (mlib_cmp (combination, ==, COMBINATION_INFINITY)) { /* Infinity */
```
With GCC 13.3.0:
```
warning: comparison of integer expressions of different signedness: ‘uint32_t’ {aka ‘unsigned int’} and ‘int’ [-Wsign-compare]
```

Similarly for `combination == COMBINATION_NAN` below on line 175.